main/pipeline/pipeline.h

```diff
@@ -48,6 +48,21 @@ size_t partitionPackageFiles(const core::GlobalState &gs, absl::Span<core::FileR
 void unpartitionPackageFiles(std::vector<ast::ParsedFile> &packageFiles,
                              std::vector<ast::ParsedFile> &&nonPackageFiles);
 
+struct CondensationLayerInfo {
+    // The `__package.rb` sources in this layer of the condensation graph.
+    absl::Span<ast::ParsedFile> packageFiles;
+
+    // The ruby sources in this layer of the condensation graph.
+    absl::Span<core::FileRef> sourceFiles;
+};
+
+// Using the condensation graph, sort the package and source files according to the layer they would show up in the
+// parallel traversal of the condensation graph. The `packageFiles` vector will be mutated to include non-test versions
```

Comment:
This "mutated to include" would go away with @aisamanra's work to split test packages out, correct?

Reply:
Yep, then the packages could be passed in as an `absl::Span` instead, as we wouldn't need to allocate any others.